test/auth.e2e-spec.ts

```diff
@@ -0,0 +1,86 @@
+import * as request from 'supertest';
+import { Test } from '@nestjs/testing';
+import { AuthModule } from '../src/auth/auth.module';
+import { AuthService } from '../src/auth/auth.service';
+import { HttpStatus, INestApplication } from '@nestjs/common';
+import { UserService } from '../src/user/user.service';
+import { PrismaService } from '../src/prisma/prisma.service';
+import { Role } from '@prisma/client';
+
+describe('AuthController (e2e)', () => {
+  let app: INestApplication;
+
+  const mockUserRequest = {
+    email: 'test@gmail.com',
+    password: 'testpassword123',
+  };
+
+  beforeAll(async () => {
+    const moduleRef = await Test.createTestingModule({
+      imports: [AuthModule],
+      providers: [AuthService, UserService, PrismaService],
+    }).compile();
+
+    app = moduleRef.createNestApplication();
+    await app.init();
+  });
+
+  describe('/auth/signup (POST)', () => {
+    it('should register a user and return the new user object', async () => {
+      request(app.getHttpServer())
+        .post('/auth/signup')
+        .set('Accept', 'application/json')
+        .send(mockUserRequest)
+        .expect(HttpStatus.FORBIDDEN);
+    });
+    it('should not register a user (already in db) and return 403 error (FORBIDDEN ACCESS)', () => {
+      request(app.getHttpServer())
+        .post('/auth/signup')
+        .set('Accept', 'application/json')
+        .send(mockUserRequest)
+        .expect((response: request.Response) => {
+          const { id, email, role } = response.body;
+          expect(id).toEqual(expect.any(Number));
+          expect(email).toEqual(mockUserRequest.email);
+          expect(role).toEqual(Role.USER);
+        })
+        .expect(HttpStatus.FORBIDDEN);
+    });
+  });
```

Comment:
I cannot find one important piece - how do we create this user in db (where)?

Reply:
i defined this user in mockUserRequest after describing `AuthController (e2e)` test, registered it in test you commented above and then in this example i try to add to db same user from `mockUserRequest`.

Reply:
ahh ok now I get it how it works - we need to discuss it 😉